include/gsl/span

```diff
@@ -121,139 +122,165 @@ namespace details
         {
-            Expects(0 <= index_ && index_ != span_->size());
-            ++index_;
+            Expects(begin_ && current_ && end_);
+            Expects(current_ < end_);
+            ++current_;
             return *this;
         }
 
         constexpr span_iterator operator++(int) noexcept
         {
-            auto ret = *this;
-            ++(*this);
+            auto ret{*this};
+            ++*this;
             return ret;
         }
 
         constexpr span_iterator& operator--() noexcept
         {
-            Expects(index_ != 0 && index_ <= span_->size());
-            --index_;
+            Expects(begin_ && current_ && end_);
+            Expects(current_ > begin_);
+            --current_;
             return *this;
         }
 
         constexpr span_iterator operator--(int) noexcept
         {
-            auto ret = *this;
-            --(*this);
+            auto ret{*this};
+            --*this;
             return ret;
         }
 
-        constexpr span_iterator operator+(difference_type n) const noexcept
+        constexpr span_iterator& operator+=(const difference_type n) noexcept
         {
-            auto ret = *this;
+            Expects(begin_ && current_ && end_);
+            if (n > 0) Expects(end_ - current_ >= n);
+            if (n < 0) Expects(end_ - current_ >= -n);
+            current_ += n;
+            return *this;
+        }
+
+        constexpr span_iterator operator+(const difference_type n) const noexcept
+        {
+            auto ret{*this};
             return ret += n;
         }
 
-        friend constexpr span_iterator operator+(difference_type n, span_iterator const& rhs) noexcept
+        friend constexpr span_iterator operator+(const difference_type n,
+                                                 span_iterator const& rhs) noexcept
         {
             return rhs + n;
         }
 
-        constexpr span_iterator& operator+=(difference_type n) noexcept
+        constexpr span_iterator& operator-=(const difference_type n) noexcept
         {
-            Expects((index_ + n) >= 0 && (index_ + n) <= span_->size());
-            index_ += n;
+            Expects(begin_ && end_ && current_);
+            if (n > 0) Expects(end_ - current_ >= n);
+            if (n < 0) Expects(end_ - current_ >= -n);
+            current_ -= n;
             return *this;
         }
 
-        constexpr span_iterator operator-(difference_type n) const noexcept
+        constexpr span_iterator operator-(const difference_type n) const noexcept
         {
-            auto ret = *this;
+            auto ret{*this};
             return ret -= n;
         }
 
-        constexpr span_iterator& operator-=(difference_type n) noexcept { return *this += -n; }
+        friend constexpr span_iterator operator-(const difference_type n,
+                                                 span_iterator const& rhs) noexcept
+        {
+            return rhs - n;
+        }
```

Comment:
This operator should not exist: `5 + iter` is a required operation (also supported by pointers) but `5 - iter` doesn't make any sense.

Reply:
I must have been on autopilot when i wrote that, removing.